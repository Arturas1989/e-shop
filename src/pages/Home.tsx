export const Home = () => {
  return (
    <main data-testid='home'>Home</main>
  )
};
