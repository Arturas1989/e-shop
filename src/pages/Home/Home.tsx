import { Hero, FeaturedProducts, Testimonials, Faq } from "./index";

export const Home = () => {
  return (
    <main data-testid='home' className="home">
      <Hero />
      <FeaturedProducts />
      <Testimonials />
      <Faq />
    </main>
  )
};
