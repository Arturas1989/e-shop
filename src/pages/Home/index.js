export { Hero } from './components/Hero/Hero';
export { FeaturedProducts } from './components/FeaturedProducts/FeaturedProducts';
export { Testimonials } from './components/Testimonials/Testimonials';