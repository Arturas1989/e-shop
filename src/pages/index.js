export { Home } from './Home/Home';
export { ProductsPage } from './Products/ProductsPage';
export { PageNotFound } from './PageNotFound/PageNotFound';

export { Register } from './Auth/Register/Register';
export { Login } from './Auth/Login/Login';
