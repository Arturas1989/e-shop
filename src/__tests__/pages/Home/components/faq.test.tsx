import renderer from 'react-test-renderer';
import { fireEvent, screen, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Faq } from '../../../../pages/Home';

describe('Faq tests', () => {
  test('render test', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Faq />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('question expand test', () => {
      render(
        <BrowserRouter>
          <Faq />
        </BrowserRouter>
      )
      const expandIcon = screen.getAllByTestId('expand-control-icon')[0];
      
      fireEvent.click(expandIcon);
      const expandedAnswer = screen.getAllByTestId('expanded-answer')[0];
      expect(expandedAnswer).toBeInTheDocument();

      fireEvent.click(expandIcon);
      expect(expandedAnswer).not.toBeInTheDocument();
  });
});