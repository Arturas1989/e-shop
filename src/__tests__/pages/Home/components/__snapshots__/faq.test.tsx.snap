// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Faq tests render test 1`] = `
<section
  className="faq"
>
  <h2
    className="heading"
  >
    Frequently asked questions
  </h2>
  <div
    className="questions"
  >
    <div
      className="question"
    >
      <div
        className="colapsed"
      >
        <p>
          Why should I use e-shop?
        </p>
        <i
          className="bi bi-chevron-down"
          data-testid="expand-control-icon"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="question"
    >
      <div
        className="colapsed"
      >
        <p>
          Can I access my e-shop on mobile?
        </p>
        <i
          className="bi bi-chevron-down"
          data-testid="expand-control-icon"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="question"
    >
      <div
        className="colapsed"
      >
        <p>
          Do you offer refunds?
        </p>
        <i
          className="bi bi-chevron-down"
          data-testid="expand-control-icon"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="question"
    >
      <div
        className="colapsed"
      >
        <p>
          Do you support International payments?
        </p>
        <i
          className="bi bi-chevron-down"
          data-testid="expand-control-icon"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</section>
`;
